workers/unity/Packages/com.improbable.gdk.playerlifecycle/Systems/PlayerCreation/HandleCreatePlayerRequestSystem.cs

```diff
@@ -12,83 +12,85 @@ namespace Improbable.Gdk.PlayerLifecycle
+            foreach (var request in requests)
+            {
+                var playerEntityTemplate = PlayerLifecycleConfig.CreatePlayerEntityTemplate(
+                    request.CallerWorkerId,
+                    request.Payload.Position
+                );
+
+                var entityRequest = new WorldCommands.CreateEntity.Request
+                (
+                    playerEntityTemplate,
+                    context: new PlayerCreationRequestContext
                     {
-                        throw new InvalidOperationException(Errors.PlayerEntityTemplateNotFound);
+                        createPlayerRequest = request
                     }
+                );
 
-                    var playerEntity = PlayerLifecycleConfig.CreatePlayerEntityTemplate(request.CallerWorkerId,
-                        request.Payload.Position);
-                    createEntitySender.RequestsToSend.Add(new WorldCommands.CreateEntity.Request
-                    (
-                        playerEntity,
-                        context: new PlayerCreationRequestContext
-                        {
-                            createPlayerRequest = request
-                        }
-                    ));
-                }
+                commandSystem.SendCommand(entityRequest);
             }
+        }
 
-            for (var i = 0; i < entityCreationResponseData.Length; ++i)
+        private void HandlePlayerCreateEntityResponses()
+        {
+            var responses = commandSystem.GetResponses<WorldCommands.CreateEntity.ReceivedResponse>();
+            foreach (var response in responses)
             {
-                var entityCreationResponses = entityCreationResponseData.CreateEntityResponses[i];
-                var responder = entityCreationResponseData.CreatePlayerResponders[i];
-
-                foreach (var receivedResponse in entityCreationResponses.Responses)
+                if (!(response.Context is PlayerCreationRequestContext requestContext))
                 {
-                    if (!(receivedResponse.Context is PlayerCreationRequestContext requestContext))
-                    {
-                        // Ignore non-player entity creation requests
-                        continue;
-                    }
+                    // Ignore non-player entity creation requests
+                    continue;
+                }
 
-                    if (receivedResponse.StatusCode != StatusCode.Success || !receivedResponse.EntityId.HasValue)
-                    {
-                        responder.ResponsesToSend.Add(new PlayerCreator.CreatePlayer
-                            .Response(requestContext.createPlayerRequest.RequestId,
-                                $"Failed to create player: \"{receivedResponse.Message}\""));
 
-                        continue;
-                    }
+                if (response.StatusCode != StatusCode.Success || !response.EntityId.HasValue)
```

Comment:
When would the status code be success but no entity id is returned?

Reply:
This is guaranteed not to happen. So to answer the todo: we can just do the status check.